src/BankAccountApp/Account.java

```diff
@@ -0,0 +1,94 @@
+// we want an abstract class here, because we'll never create an Account object
+// we'll be creating BankAccountApp.Checking and BankAccountApp.Savings objects (that extend Account)
+
+package BankAccountApp;
+
+
+import java.util.concurrent.ThreadLocalRandom;
+
+public abstract class Account implements IRate {
+    // list common properties for savings and checking account
+    private String name;
+    private String sSN;
+
+    double initDeposit;
+    private double balance;
+    private static int index = 10000;  // base of account number, we will increment each time we create a new account
```

Comment:
The `static` keyword definitely works to keep the base account number incrementing between invocations of `Account()` but in general, this is a little frowned upon. It's not the `Account`'s job to keep track of the base account number, that's the users' job (`main` in this case). You could switch this from a `static` field to a constructor parameter